course/views.py

```diff
@@ -101,3 +101,27 @@ def list(self, request):
             queryset = queryset.filter(course=course_id)
         serializer = self.get_serializer_class()(queryset, many=True)
         return Response(serializer.data)
+    
+    def perform_destroy(self, instance):
+        """
+        override perform_destroy to refund balance from enrollment to parent
+        """
+
+        # refund parent remaining enrollment balance
+        if instance.student.primary_parent:
+            parent = instance.student.primary_parent
+        else:
+            parent = instance.student.secondary_parent
+        parent.balance += instance.enrollment_balance
+        parent.save()
+
+        # remove enrollment
+        deleted_instance = super().perform_destroy(instance)
+
+        # remove course if this is last enrollment
+        if len(instance.course.enrollment_id_list) == 0:
```

Comment:
Let's not do this actually - If this was a class, then if the first person to sign up requests a refund, it would delete the class

Reply:
We don't need to do this here then.

However, the implementation of course destroy does not take into account the PROTECTED references to it from Sessions and Enrollments.
Course model destroy() function must be overwritten to first remove these foreign references.  

should make sure to delete all enrollments and sessions